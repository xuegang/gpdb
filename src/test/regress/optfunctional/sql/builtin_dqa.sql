-- start_ignore
create schema builtin_dqa;
set search_path to builtin_dqa;
CREATE TABLE foo (a text, b int, c text, d text, e text);
INSERT INTO foo VALUES ('foo', 12000, 'pg_default', 'ao_tab1', 'testdb');
INSERT INTO foo VALUES ('bar', 4045, 'pg_default', 'ao_tab2', 'testdb');
INSERT INTO foo VALUES ('ao_tab1', 573, 'pg_default', 'ao_tab1', 'testdb');
INSERT INTO foo VALUES ('ao_tab2', 25730, 'pg_default', 'ao_tab2', 'testdb');
-- end_ignore

-- @description varying_with_generate_series_clock_timestamp.sql
-- @db_name builtin_functionproperty
-- @author tungs1
-- @modified 2013-04-17 12:00:00
-- @created 2013-04-17 12:00:00
-- @executemode ORCA_PLANNER_DIFF
-- @tags functionPropertiesBuiltin HAWQ
SELECT count( distinct i ) FROM (SELECT clock_timestamp() i, generate_series(1, 5) j) t;
-- @description varying_with_generate_series_random.sql
-- @db_name builtin_functionproperty
-- @author tungs1
-- @modified 2013-04-17 12:00:00
-- @created 2013-04-17 12:00:00
-- @executemode ORCA_PLANNER_DIFF
-- @tags functionPropertiesBuiltin HAWQ
SELECT count( distinct i ) FROM (SELECT random() i, generate_series(1, 5) j) t;
-- @description varying_with_generate_series_timeofday.sql
-- @db_name builtin_functionproperty
-- @author tungs1
-- @modified 2013-04-17 12:00:00
-- @created 2013-04-17 12:00:00
-- @executemode ORCA_PLANNER_DIFF
-- @tags functionPropertiesBuiltin HAWQ
SELECT count( distinct i ) FROM (SELECT timeofday() i, generate_series(1, 5) j) t;
-- @description varying_fromtable_clock_timestamp.sql
-- @db_name builtin_functionproperty
-- @author tungs1
-- @modified 2013-04-17 12:00:00
-- @created 2013-04-17 12:00:00
-- @executemode gt1
-- @tags functionPropertiesBuiltin HAWQ
SELECT count( distinct i ) > 1 AS gt1 FROM (SELECT clock_timestamp() i from foo) t;
-- @description varying_fromtable_random.sql
-- @db_name builtin_functionproperty
-- @author tungs1
-- @modified 2013-04-17 12:00:00
-- @created 2013-04-17 12:00:00
-- @executemode gt1
-- @tags functionPropertiesBuiltin HAWQ
SELECT count( distinct i ) > 1 AS gt1 FROM (SELECT random() i from foo) t;
-- @description varying_fromtable_timeofday.sql
-- @db_name builtin_functionproperty
-- @author tungs1
-- @modified 2013-04-17 12:00:00
-- @created 2013-04-17 12:00:00
-- @executemode gt1
-- @tags functionPropertiesBuiltin HAWQ
SELECT count( distinct i ) > 1 AS gt1 FROM (SELECT timeofday() i from foo) t;

-- start_ignore
drop schema builtin_dqa cascade;
-- end_ignore
