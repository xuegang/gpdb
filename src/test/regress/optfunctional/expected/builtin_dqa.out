-- start_ignore
create schema builtin_dqa;
set search_path to builtin_dqa;
CREATE TABLE foo (a text, b int, c text, d text, e text);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'a' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
INSERT INTO foo VALUES ('foo', 12000, 'pg_default', 'ao_tab1', 'testdb');
INSERT INTO foo VALUES ('bar', 4045, 'pg_default', 'ao_tab2', 'testdb');
INSERT INTO foo VALUES ('ao_tab1', 573, 'pg_default', 'ao_tab1', 'testdb');
INSERT INTO foo VALUES ('ao_tab2', 25730, 'pg_default', 'ao_tab2', 'testdb');
-- end_ignore
-- @description varying_with_generate_series_clock_timestamp.sql
-- @db_name builtin_functionproperty
-- @author tungs1
-- @modified 2013-04-17 12:00:00
-- @created 2013-04-17 12:00:00
-- @executemode ORCA_PLANNER_DIFF
-- @tags functionPropertiesBuiltin HAWQ
SELECT count( distinct i ) FROM (SELECT clock_timestamp() i, generate_series(1, 5) j) t;
 count 
-------
     1
(1 row)

-- @description varying_with_generate_series_random.sql
-- @db_name builtin_functionproperty
-- @author tungs1
-- @modified 2013-04-17 12:00:00
-- @created 2013-04-17 12:00:00
-- @executemode ORCA_PLANNER_DIFF
-- @tags functionPropertiesBuiltin HAWQ
SELECT count( distinct i ) FROM (SELECT random() i, generate_series(1, 5) j) t;
 count 
-------
     1
(1 row)

-- @description varying_with_generate_series_timeofday.sql
-- @db_name builtin_functionproperty
-- @author tungs1
-- @modified 2013-04-17 12:00:00
-- @created 2013-04-17 12:00:00
-- @executemode ORCA_PLANNER_DIFF
-- @tags functionPropertiesBuiltin HAWQ
SELECT count( distinct i ) FROM (SELECT timeofday() i, generate_series(1, 5) j) t;
 count 
-------
     1
(1 row)

-- @description varying_fromtable_clock_timestamp.sql
-- @db_name builtin_functionproperty
-- @author tungs1
-- @modified 2013-04-17 12:00:00
-- @created 2013-04-17 12:00:00
-- @executemode gt1
-- @tags functionPropertiesBuiltin HAWQ
SELECT count( distinct i ) > 1 AS gt1 FROM (SELECT clock_timestamp() i from foo) t;
 gt1 
-----
 t
(1 row)

-- @description varying_fromtable_random.sql
-- @db_name builtin_functionproperty
-- @author tungs1
-- @modified 2013-04-17 12:00:00
-- @created 2013-04-17 12:00:00
-- @executemode gt1
-- @tags functionPropertiesBuiltin HAWQ
SELECT count( distinct i ) > 1 AS gt1 FROM (SELECT random() i from foo) t;
 gt1 
-----
 t
(1 row)

-- @description varying_fromtable_timeofday.sql
-- @db_name builtin_functionproperty
-- @author tungs1
-- @modified 2013-04-17 12:00:00
-- @created 2013-04-17 12:00:00
-- @executemode gt1
-- @tags functionPropertiesBuiltin HAWQ
SELECT count( distinct i ) > 1 AS gt1 FROM (SELECT timeofday() i from foo) t;
 gt1 
-----
 t
(1 row)

-- start_ignore
drop schema builtin_dqa cascade;
NOTICE:  drop cascades to table foo
-- end_ignore
